opensearch/src/main/java/org/opensearch/sql/opensearch/monitor/OpenSearchResourceMonitor.java

```diff
@@ -36,6 +36,15 @@ public OpenSearchResourceMonitor(Settings settings, OpenSearchMemoryHealthy memo
             .ignoreExceptions(OpenSearchMemoryHealthy.MemoryUsageExceedFastFailureException.class)
             .build();
     retry = Retry.of("mem", config);
+    retry
+        .getEventPublisher()
+        .onRetry(
+            event -> {
+              if (event.getNumberOfRetryAttempts() == 1) {
+                System.gc();
```

Comment:
My concern is that triggering System.gc() may impact other OpenSearch operations, such as indexing.

Reply:
> Just curious what's the behavior of not explicitly calling System.gc()?

The default GC runs async in the background, if you're still initializing a lot of objects on a smaller-resource cluster you can end up running out of memory before GC finishes (especially if running over older GC generations). Synchronously calling it has the benefit that you actually wait for the GC to finish before the retry.